__tests__/auth.test.ts

```diff
@@ -1,30 +1,288 @@
-import request from 'supertest'
-import { expect, describe, test } from 'vitest'
-import { gqlApiUrl } from './testSetup.test.js'
-import { searchSubmissionsQuery } from './submissions.test.js'
-import { Submission, SubmissionSearchFilters } from '../src/typeDefs/gqlTypes.js'
-import { gqlRequest } from '../utils/gqlTool.js'
-
-describe('auth', () => {
-    test('can login and access secure routes', async () => {
-        // TODO add auth0 test here
-
-        // test a secure endpoint as a logged in user
-        const searchSubmissionsRequest = {
-            query: searchSubmissionsQuery,
-            variables: {
-                filters: {} satisfies SubmissionSearchFilters
-            }
-        } as gqlRequest
-
-        //Get the submissions query data
-        const searchResult = await request(gqlApiUrl).post('').send(searchSubmissionsRequest)
-
-        //should not have errors
-        expect(searchResult.body.errors).toBeUndefined()
-
-        const searchedSubmissions = searchResult.body.data.submissions as Submission[]
-        
-        expect(searchedSubmissions).toBeDefined()
+import { describe, it, expect, beforeEach, vi } from 'vitest'
+import { authorize, buildUserContext, Role, Scope } from '../src/auth'
+import { envVariables } from '../utils/environmentVariables'
+import { FastifyRequest } from 'fastify'
+
+// Mock envVariables to control testing environment
+beforeEach(() => {
+    vi.restoreAllMocks()
+})
+
+describe('authorize()', () => {
```

Comment:
Let's talk on slack, but I think we should reduce these tests down to three tests. And we should also change these to hit the endpoint, not the authorize method. 

1) Hit an endpoint that has auth and successfully validates a user. 
2) Hit an endpoint that has auth and successfully denies an unauthorized user. 
3) Your last test is good to keep to validate that the payload matches our expected format. This is a nice test because that is an external dependency that we can ensure is working + hasn't changed which would break our code. These are usually called dependency tests. 

This will test the endpoint end-to-end works, validate the `authorize` method works, and ensure the end user's experience is as expected.